src/routes/write-exercise/WriteExerciseScreen.tsx

```diff
@@ -44,33 +45,33 @@ const WriteExerciseScreen = ({ navigation, route }: WriteExerciseScreenPropsType
   const { trainingSet, trainingSetId, disciplineTitle } = extraParams
   const [isLoading, setIsLoading] = useState(true)
   const [currentDocumentNumber, setCurrentDocumentNumber] = useState(0)
+  const [newDocuments, setNewDocuments] = useState<DocumentType[] | null>(null)
+  const response = useLoadDocuments(trainingSetId)
 
-  let documents = useLoadDocuments(trainingSetId).data
+  const documents = newDocuments ?? retryData?.data ?? response.data
 
   useEffect(() => {
     AsyncStorage.setSession(route.params).catch(e => console.error(e))
   }, [route.params])
 
-  if (retryData) {
-    documents = retryData.data
-  }
-
   const tryLater = (): void => {
     if (documents !== null) {
       const currDocument = documents[currentDocumentNumber]
       const newDocuments = documents.filter(d => d !== currDocument)
       newDocuments.push(currDocument)
-      documents = newDocuments
+      setNewDocuments(newDocuments)
```

Comment:
should be a `useCallback` shouldn't it?

Reply:
@SteffiStuffel what do you think?

Reply:
You mean the hole tryLater-function? Yes, that would make sense.